fixit/cli/run_rules.py

```diff
@@ -87,33 +100,48 @@ def get_formatted_reports_for_path(
     return [opts.formatter.format(rr) for rr in raw_reports]
 
 
-def main(raw_args: Sequence[str]) -> int:
-    parser = argparse.ArgumentParser(
-        description=(
-            "Validates your lint rules by running them against the specified, "
-            + "directory or file(s). This is not a substitute for unit tests, "
-            + "but it can provide additional confidence in your lint rules.\n"
-            + "If no lint rules or packages are specified, runs all lint rules "
-            + "found in the packages specified in `fixit.config.yaml`."
-        ),
-        parents=[
-            get_paths_parser(),
-            get_rules_parser(),
-            get_use_ignore_comments_parser(),
-            get_skip_ignore_byte_marker_parser(),
-            get_compact_parser(),
-            get_multiprocessing_parser(),
-        ],
-    )
-
+def _parser_arguments(
+    parser: Union[argparse._SubParsersAction, argparse.ArgumentParser],
+    sub_parser: bool = True,
+) -> None:
+    """All required arguments for `run_rules`"""
+    # pyre-ignore Pyre confused when the argument type is two
     parser.add_argument(
         "--cache-timeout",
         type=int,
         help="Timeout (seconds) for metadata cache fetching. Default is 2 seconds.",
         default=2,
     )
 
-    args = parser.parse_args(raw_args)
+    if sub_parser:
+        # pyre-ignore Pyre confused when the argument type is two
+        parser.set_defaults(subparser_fn=_main)
+    else:
+        # pyre-ignore Pyre confused when the argument type is two
+        _main(parser.parse_args())
+
+
+# pyre-ignore Pyre doesn't know about varible type when None given
+def register_subparser(parser: argparse._SubParsersAction = None) -> None:
+    """Add parser or subparser for `run_rules` command."""
+    if parser is None:
+        run_rules_parser = argparse.ArgumentParser(
+            description=DESCRIPTION,
+            parents=PARENTS,
+        )
+        _parser_arguments(run_rules_parser, sub_parser=False)
+
+    else:
+        run_rules_parser = parser.add_parser(
+            "run_rules",
+            description=DESCRIPTION,
+            parents=PARENTS,
+            help="Run fixit rules against python code",
+        )
+        _parser_arguments(run_rules_parser)
```

Comment:
Move the logic to register_subparser and rename `_parser_arguments` as `_add_arguments` can keep the `_add_arguments` focuses on adding arguments as well as removing the type check.

```suggestion
def register_subparser(parser: argparse._SubParsersAction = None) -> None:
    """Add parser or subparser for `run_rules` command."""
    if parser is None:
        run_rules_parser = argparse.ArgumentParser(
            description=DESCRIPTION,
            parents=PARENTS,
        )
        _parser_arguments(run_rules_parser, sub_parser=False)
        _main(run_rules_parser.parse_args())
    else:
        run_rules_parser = parser.add_parser(
            "run_rules",
            description=DESCRIPTION,
            parents=PARENTS,
            help="Run fixit rules against python code",
        )
        _parser_arguments(run_rules_parser)
        run_rules_parser.set_defaults(subparser_fn=_main)
```